src/connection.js

```diff
@@ -22,8 +23,9 @@ class Connection extends EventEmitter {
 
   getHandle () {
     return this.handlePromise || (this.handlePromise = new Promise((resolve, reject) => {
-      const { host, port } = this.options
-      const socket = createConnection(port, host)
+      const socket = this.options.tls
+        ? createSecureConnection(this.options)
```

Comment:
first of all, thanks for clean and concise PR!
regarding this and following line, I would prefer explicit specification which properties of `options` object we pass to createSecureConnection/createConnection

Reply:
You're welcome :) For TLS connections to some servers (especially compose.io rabbitmq), it's needed to pass the `servername` to with the connection parameters instead of just the host and the port. That's the reason why I passed the whole options object to the `createConnection` method.

Reply:
the single reason why I preferred to take `connect(port, host)` version of method is that [servername is required to be dns name, not direct address](https://nodejs.org/api/tls.html#tls_tls_connect_options_callback)

so in case of `amqps://user:pass@127.0.0.1/vhost` if we pass { ..., servername: '127.0.0.1' } I was not sure what will be the result

I'll setup test env and try this today, if it works, I'll do minor release

Reply:
@maxnowack please check if 1.4.0 works for your setup - I've added `servername` option to connection method
P. S. sorry for such a long delay